### PR TITLE
refactor(app): use spinnermodalpage in calcheck

### PR DESCRIPTION
**Pull request by sfoster1**
This makes the look the same as deck calibration.

## Risk Assessment

Low, it's about the look in cal check only.

### CODE REVIEW

**Review by IanLondon — APPROVED** [review]
looks OK to me

merge conflict for this PR though

File: app/src/components/CheckCalibration/index.js [hunk 1]
```diff
@@ -325,27 +326,29 @@ export function CheckCalibration(props: CheckCalibrationProps): React.Node {
       break
     }
     default: {
-      stepContents = <SpinnerModal />
+      showSpinner = true
```
Comment (mcous) [inline app/src/components/CheckCalibration/index.js:329]: Could this be an early return before the switch case instead?

```js
if (pending) {
  return <SpinnerModalPage titleBar={titleBarProps} />
}

// ...
switch (currentStep) {
```

File: app/src/components/CheckCalibration/index.js [hunk 2]
```diff
@@ -325,27 +326,29 @@ export function CheckCalibration(props: CheckCalibrationProps): React.Node {
...
     }
   }
-
-  return (
+  const titleBarProps = {
```
Comment (mcous) [inline app/src/components/CheckCalibration/index.js:332]: Could you define this up top with the rest of the variables to make the `shouldDisplayTitleBarExit` context a little easier to follow? This component is pretty drastically large as is

**Comment by mcous** [comment]
Oh, also, I don't see any tests at all for "display a spinner while request is pending", which seems like an important part of this UI

File: app/src/components/CheckCalibration/index.js (hunk 2 above)
Comment (sfoster1) [reply app/src/components/CheckCalibration/index.js:332]: It needs to be defined low enough to catch the modifications to shouldDisplayTitleBarExit, though. I could make it a builder function that takes that as a param?

File: app/src/components/CheckCalibration/index.js (hunk 1 above)
Comment (sfoster1) [reply app/src/components/CheckCalibration/index.js:329]: As currently implemented it's also the default case for undhandled step in the `default` tag which I guess doesn't really need to be true

File: app/src/components/CheckCalibration/index.js (hunk 2 above)
Comment (mcous) [reply app/src/components/CheckCalibration/index.js:332]: Doh. My bad, carry on

**Review by mcous — APPROVED** [review]

**Pull request merged**